# Step 1: Genotype File and Format - Script Outline

Here raw genotype data was acquired from UKBB that represent similar GWAS significant HCM gene variants identified in our groups collaborative study. This data comprised two files; an SNP list file (.txt) with the risk allele information (for BAG3 (chr10, rs2234962) and FHOD3 (chr18, rs2303510)); and the corresponding sample genotype file (.ped) (comprising allelic information such as TT, CT, TC, or CC for BAG3 or GG, AG, GA or AA for FHOD3). An additive parameter was chosen to represent this genetic model, and hence the genotype data was converted into a binary numeric format for processability and accuracy. Where, for BAG3, 0 = TT, 1 = CT/TC and 2 = CC (Additive for C) and for FODH3, 0 = GG, 1 = AG/GA and 2 = AA (Additive for A). The data was then re-formatted with missing data rows removed, relevant sample identification (ID) and genotype columns selected, and the variable columns renamed. Various quality control (QC) measures were undertaken to ensure data could be further processed in R. 
